lambdas/services/base/dynamo_service.py

```diff
@@ -122,6 +122,34 @@ def update_item(self, table_name: str, key: str, updated_fields: dict):
             ExpressionAttributeValues=expression_attribute_values,
         )
 
+    def update_conditional(
+        self,
+        table_name: str,
+        key: str,
+        updated_fields: dict,
+        condition_expression: str,
+        expression_attribute_values,
+    ):
+        table = self.get_table(table_name)
+
+        updated_field_names = list(updated_fields.keys())
+        update_expression = create_update_expression(updated_field_names)
+        _, expression_attribute_names = create_expressions(updated_field_names)
+
+        expression_attribute_values.update(
+            create_expression_attribute_values(updated_fields)
+        )
+
+        update_item_args = {
+            "Key": {"ID": key},
+            "UpdateExpression": update_expression,
+            "ExpressionAttributeNames": expression_attribute_names,
+            "ExpressionAttributeValues": expression_attribute_values,
+        }
+        update_item_args["ConditionExpression"] = condition_expression
+
+        table.update_item(**update_item_args)
+
```

Comment:
Maybe expand `update_item` to accept `ConditionExpression` instead of a new function?